internal/kms/rewrapping.go

```diff
@@ -0,0 +1,26 @@
+package kms
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"golang.org/x/exp/maps"
+)
+
+type RewrapFn func(ctx context.Context, dataKeyVersionId string, scopeId string, reader db.Reader, writer db.Writer, kms *Kms) error
+
+var tableNameToRewrapFn = map[string]RewrapFn{}
+
+// RegisterTableRewrapFn registers a function to be used to rewrap data in a specific table with a new key
+func RegisterTableRewrapFn(tableName string, rewrapFn RewrapFn) {
```

Comment:
should we have unit tests for this func?

Reply:
Added a basic test for this